Automattic/wp-calypso
# Add QR code in post action ellipsis menu

#### Proposed Changes

* Add an option to generate a QR code in `Posts` and `Pages` lists, specifically in the ellipsis menu.
* This feature will be disabled in `production` for now, since the feature is behind a feature flag. It will be available on local and staging environments.

#### Screenshots
Posts list | Pages list | QR Modal
---|---|---
<img width="222" alt="image" src="https://user-images.githubusercontent.com/1488641/199628036-4a1858b9-7a39-4352-afca-3aa0211e2547.png"> | <img width="211" alt="image" src="https://user-images.githubusercontent.com/1488641/199628250-1aff4e93-68cf-4d06-bade-f301701c9555.png"> | <img width="257" alt="image" src="https://user-images.githubusercontent.com/1488641/199628197-9ee37f99-a65c-4445-b1ae-606e47ce57f4.png">

#### Testing Instructions



1. Clone the repository and checkout to this branch, see `GET STARTED` on the README section of this repo.
2. Start Calypso blue `yarn start`
3. Navigate to `http://calypso.localhost:3000/`
4. Switch to your site and navigate to `Posts` section.
5. Click on the ellipsis menu (the tree dots) on a published post, and you should see the new QR Code option.
6. Click on `QR Code` button and a modal dialog will be displayed with the QR code of the post selected.
7. Check with your phone if the QR code is pointing to the expected URL.
8. Do step 5, 6 and 7 again on `Pages` section.

**Testing feature enablement**
This feature will not be available in production yet, so we need to ensure that this feature is only enable where we config `post-list/qr-code-link` to `true`. For now, the feature will be available on the development and staging environments (proxied). 
* Make sure you see the QR code option in the post list dropdown.
* Go to your development.json file and change the value of `post-list/qr-code-link` to false.
* Ctrl+C your `yarn start` command and restart it.
* Make sure you don't see the QR code option in the post list dropdown, since the feature flag is turned off.

#### Pre-merge Checklist
